File: rust/wikibooks-rust-learning-by-making/ch2/sum_textfile.rs
use std::{env, fs};

fn main() {
  let args = env::args();

  let mut total: f64 = 0.0;

  for (i, filename) in args.enumerate() {
    if i == 0 { continue; }

    let text = fs::read_to_string(filename).unwrap();

    let lines = text.split('\n');

    for line in lines {
      let n: f64 = match line.parse() {
        Ok(v) => v,
        Err(_) => 0.0,
      };
      total += n;
    }
  }

  println!("{}", total);
}

File: rust/wikibooks-rust-learning-by-making/ch2/sum_textfile.py
import sys

total = 0

for i, v in enumerate(sys.argv):
    if i == 0:
        continue

    with open(v, "rt") as fp:
        text = fp.read()

        for line in text.split("\n"):
            try:
                total += float(line)
            except ValueError:
                pass

print(total)
